File: CodeLocatorApp/CodeLocatorCore/src/main/java/com/bytedance/tools/codelocator/operate/Operate.java
package com.bytedance.tools.codelocator.operate;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.bytedance.tools.codelocator.model.OperateData;
import com.bytedance.tools.codelocator.model.ResultData;

/**
 * Created by liujian.android on 2024/4/2
 *
 * @author liujian.android@bytedance.com
 */
public abstract class Operate {

    public abstract @NonNull String getOperateType();

    public boolean executeCommandOperate(@NonNull Activity activity, @NonNull OperateData operateData, @NonNull ResultData result) {
        return false;
    }

}

File: CodeLocatorApp/CodeLocatorCore/src/main/java/com/bytedance/tools/codelocator/operate/Operate.kt
package com.bytedance.tools.codelocator.operate

import android.app.Activity
import com.bytedance.tools.codelocator.model.OperateData
import com.bytedance.tools.codelocator.model.ResultData

abstract class Operate {

    abstract fun getOperateType(): String

    abstract fun excuteCommandOperate(activity: Activity, operateData: OperateData): Boolean

    open fun excuteCommandOperate(activity: Activity, operateData: OperateData, result: ResultData): Boolean {
        return excuteCommandOperate(activity, operateData)
    }

    fun excuteCommand(activity: Activity, operateData: OperateData, result: ResultData): Boolean {
        return excuteCommandOperate(activity, operateData, result)
    }

}